apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryInsertResults.cs

```diff
@@ -135,14 +136,28 @@ public BigQueryInsertResults ThrowOnAnyError()
                 return this;
             }
 
-            var exception = new GoogleApiException(_client.Service.Name, $"Error inserting data. Status: {Status}")
+            // If we attempted to insert just one row, let's be more explicit about the error message.
```

Comment:
Any reason this is only if we're inserting a single row? Is that because if we've got a lot of errors, we don't want the ToString representation to be huge? It feels a little inconsistent that client code can't use the Error property if they're inserting more than one row... but we may need to accept that as a practical trade-off.

Reply:
(Perhaps we could create a GoogleApiException containing the first error unconditionally, along with the number of errors, and make the message explain that the caller should inspect the BigQueryInsertResult for more information?)

Reply:
Client code can always use the Error property, the only difference here is what the Error.Message is (which will be part of the GoogleApiException.Message). If we inserted only one row, we can use one error message for that row as the exception message, but if we inserted many and many failed then the exception message could be huge.

I can actually change this to include more cases, for instance, even if we inserted many rows, if just one row failed, etc.

I'll also make the code clearer so that is very explicit that the only thing different is the Error.Message.

Reply:
Right - and the one time that client code *can't* use the error property is because they've called InsertRow which automatically throws, and in that case they'll have the error here.

I think unconditionally including the first error but indicating how many total errors there are could be really useful.